spinn_machine/sdram.py

```diff
@@ -17,6 +17,7 @@ def __init__(self, size=DEFAULT_SDRAM_BYTES):
         if size < 0:
             raise SpinnMachineInvalidParameterException(
                 "size", size, "negative sizes are meaningless")
+        SDRAM.DEFAULT_SDRAM_BYTES = size
```

Comment:
That's really odd. While it's great that we're able to do `SDRAM(size=real_size)`, it's not at all great that a subsequent `SDRAM()` would merrily undo that.